frontend/src/app/guards/admin-guard/admin-guard.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Router, RouterStateSnapshot } from '@angular/router';
+import { concatAll, Observable, of } from 'rxjs';
+import { AuthRole } from 'src/app/enums/auth-role-enum';
+import { NotificationText } from 'src/app/enums/notification-text-enum';
+import { RouteName } from 'src/app/enums/route-name-enum';
+import { AuthService } from 'src/app/services/auth/auth.service';
+import { NotificationService } from 'src/app/services/notification/notification.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminGuardService {
+  private isAdminRole: boolean = false;
+
+  constructor(
+    private authService: AuthService, 
+    private router: Router,
+    private notificationService: NotificationService,
+  ) { }
+
+  public canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
```

Comment:
If i understand correctly this method can be written shorter

if(!this.authService.role.getValue() === AuthRole.ADMIN || !this.authService.checkTokenExpire()) {
      this.router.navigate([RouteName.AUTH]);
      this.notificationService.showError(NotificationText.FORBIDDEN, 3000);

      return of(false);
}

return of(true);